test/cmd/builds.sh

```diff
@@ -12,16 +12,64 @@ os::log::install_errexit
 url=":${API_PORT:-8443}"
 project="$(oc project -q)"
 
+# expect_build_output_spec runs `oc new-build` with given arguments and expects
+# a build config with a given name to be created with a given build output spec
+function os::cmd::builds::expect_build_output_spec() {
+	if [[ $# -ne 3 ]]; then echo "os::cmd::builds::expect_build_output_spec expects three argumments, got $#"; fi
+	local newbuild_args="$1"
+  local expected_buildconfig_name="$2"
+	local expected_output_spec="$3"
+
+  local template='{{with .spec.output.to}}{{.kind}} {{.name}}{{end}}'
+	local get_output_spec_cmd="oc get bc/${expected_buildconfig_name} --template '${template}'"
+
+	os::cmd::expect_success 'oc delete all --all'
+	os::cmd::expect_success "oc new-build ${newbuild_args}"
+	os::cmd::expect_success_and_text "${get_output_spec_cmd}" "^${expected_output_spec}\$"
+}
+
 # This test validates builds and build related commands
 
 os::cmd::expect_success 'oc new-build openshift/ruby-20-centos7 https://github.com/openshift/ruby-hello-world.git'
 os::cmd::expect_success 'oc get bc/ruby-hello-world'
 os::cmd::expect_success 'cat "${OS_ROOT}/Dockerfile" | oc new-build -D - --name=test'
 os::cmd::expect_success 'oc get bc/test'
-os::cmd::expect_success "oc new-build --dockerfile=\$'FROM centos:7\nRUN yum install -y httpd'"
-os::cmd::expect_success 'oc get bc/centos'
-os::cmd::expect_success 'oc delete all --all'
 
+# Build from Dockerfile with output to ImageStreamTag
+os::cmd::builds::expect_build_output_spec \
+	"--dockerfile=\$'FROM centos:7\nRUN yum install -y httpd'" \
+	'centos' \
+	'ImageStreamTag centos:latest'
+
+# Build from Dockerfile with output to DockerImage
+os::cmd::builds::expect_build_output_spec \
+	"-D \$'FROM openshift/origin:v1.1' --to-docker" \
+	'origin' \
+	'DockerImage library/origin:latest'
+
+# Build from Dockerfile with given output ImageStreamTag spec
+os::cmd::builds::expect_build_output_spec \
+	"-D \$'FROM openshift/origin:v1.1\nENV ok=1' --to origin-test:v1.1" \
+	'origin-test' \
+	'ImageStreamTag origin-test:v1.1'
+
+# Build from Dockerfile with given output DockerImage spec
+os::cmd::builds::expect_build_output_spec \
+	"-D \$'FROM openshift/origin:v1.1\nENV ok=1' --to-docker --to openshift/origin:v1.1-test" \
+	'origin' \
+	'DockerImage openshift/origin:v1.1-test'
+
+# Build from Dockerfile with custom name and given output ImageStreamTag spec
+os::cmd::builds::expect_build_output_spec \
+	"-D \$'FROM openshift/origin:v1.1\nENV ok=1' --to origin-name-test --name origin-test" \
+	'origin-test' \
+	'ImageStreamTag origin-name-test:latest'
+
+os::cmd::expect_failure_and_text \
+	'oc new-build centos/ruby-22-centos7~https://github.com/openshift/ruby-ex centos/php-56-centos7~https://github.com/openshift/cakephp-ex --to invalid/argument' \
+	'error: only one component or source repository can be used when specifying an output image reference'
```

Comment:
@smarterclayton @bparees also included this test to check "--to" given two source repos (invalid usage).